geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java

```diff
@@ -0,0 +1,107 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ *************************************************************************/
+
+package org.locationtech.geomesa.hbase.filters;
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.exceptions.DeserializationException;
+import org.apache.hadoop.hbase.filter.FilterBase;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.filter.text.ecql.ECQL;
+import org.locationtech.geomesa.features.interop.SerializationOptions;
+import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer;
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+import java.io.IOException;
+
+public class JSimpleFeatureFilter extends FilterBase {
+    private String sftString;
+    private SimpleFeatureType sft;
+    private KryoFeatureSerializer serializer;
+
+    private org.opengis.filter.Filter filter;
+    private String filterString;
+
+    public JSimpleFeatureFilter(String sftString, String filterString) {
+        this.sftString = sftString;
+        sft = SimpleFeatureTypes.createType("", sftString);
+        serializer = new KryoFeatureSerializer(sft, SerializationOptions.withoutId());
+
+        this.filterString = filterString;
+        if (filterString != null && filterString != "") {
```

Comment:
If the filterString is empty, filter won't be set on line 41 and will remain null. Then on line 57 the null check will pass and it will simply include that key value pair. It shouldn't reach line 61.

Is there something wrong with my understanding?